lib/pdk/cli/util.rb

```diff
@@ -144,6 +144,7 @@ def puppet_from_opts_or_env(opts, logging_disabled = false)
         begin
           puppet_env =
             if use_puppet_dev
+              PDK::Util::PuppetVersion.fetch_puppet_dev
```

Comment:
Should this be guarded on only once per session? Just incase this method is called mulitple times?

Reply:
Fixed by @rodjek